client.go

```diff
@@ -115,9 +124,41 @@ func (c *Client) request(ctx context.Context, method string, urlStr string, quer
 	req.Header.Add("Notion-Version", c.notionVersion)
 	req.Header.Add("Content-Type", "application/json")
 
-	res, err := c.httpClient.Do(req.WithContext(ctx))
-	if err != nil {
-		return nil, err
+	failedAttempts := 0
+	var res *http.Response
+	for {
+		res, err := c.httpClient.Do(req.WithContext(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		if res.StatusCode == http.StatusTooManyRequests {
+			failedAttempts++
+			if failedAttempts == 3 {
+				return nil, errors.New("Retry request with 429 response failed")
+			}
+			var delay time.Duration
+			// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After
+			retryAfter := res.Header["Retry-After"][0]
+			waitUntil, err := time.Parse(time.RFC1123, retryAfter)
+			if err != nil {
+				waitSeconds, err := strconv.Atoi(retryAfter)
+				if err != nil {
+					break // should not happen
+				} else {
+					delay = time.Duration(waitSeconds) * time.Second
+				}
+			} else {
+				delay = waitUntil.Sub(time.Now())
+			}
```

Comment:
IIUC, the intention here is to try parsing the Retry-After value as a date&time, and then try parsing it as an integer (number of seconds to wait), since both options are mentioned on link you added.

In the Notion API reference ([link](https://developers.notion.com/reference/request-limits#rate-limits)) though it's mentioned that the header contains the number of seconds to wait. With that, I think this code can be simplified.

Reply:
Ah, good observation!